src/sdk/pynni/nni/metis_tuner/metis_tuner.py

```diff
@@ -121,13 +121,12 @@ def update_search_space(self, search_space):
                 key_range = search_space[key]['_value']
                 idx = self.key_order.index(key)
                 if key_type == 'quniform':
-                    if key_range[2] == 1:
-                        self.x_bounds[idx] = [key_range[0], key_range[1]]
+                    if key_range[2] == 1 and key_range[0].is_interger() and key_range[1].is_interger():
+                        self.x_bounds[idx] = [key_range[0], key_range[1]+1]
                         self.x_types[idx] = 'range_int'
                     else:
-                        bounds = []
-                        for value in np.arange(key_range[0], key_range[1], key_range[2]):
-                            bounds.append(value)
+                        low, high, q = key_range
```

Comment:
so `q` must be an integer?

Reply:
no, why this question ?

Reply:
because the type in line 131 is `discrete_int`

Reply:
"discreate_int" doesn't mean that the value will be int. here is to deal with the index of the generated list